friends_3ds.go

```diff
@@ -47,33 +59,49 @@ const (
 
 // Friends3DSProtocol handles the Friends (3DS) nex protocol
 type Friends3DSProtocol struct {
-	server                         *nex.Server
-	UpdateProfileHandler           func(err error, client *nex.Client, callID uint32, profileData *MyProfile)
-	UpdateMiiHandler               func(err error, client *nex.Client, callID uint32, mii *Mii)
-	UpdatePreferenceHandler        func(err error, client *nex.Client, callID uint32, unknown1 bool, unknown2 bool, unknown3 bool)
-	UpdatePresenceHandler          func(err error, client *nex.Client, callID uint32, presence *NintendoPresence, unknown bool)
-	UpdateFavoriteGameKeyHandler   func(err error, client *nex.Client, callID uint32, gameKey *GameKey)
-	UpdateCommentHandler           func(err error, client *nex.Client, callID uint32, comment string)
-	SyncFriendHandler              func(err error, client *nex.Client, callID uint32, unknown1 uint64, unknown2 []uint32, unknown3 []uint64)
-	AddFriendByPrincipalIDHandler  func(err error, client *nex.Client, callID uint32, unknown1 uint64, principalID uint32)
-	GetFriendPersistentInfoHandler func(err error, client *nex.Client, callID uint32, pidList []uint32)
-	GetFriendMiiHandler            func(err error, client *nex.Client, callID uint32, pidList []uint32)
-	GetFriendPresenceHandler       func(err error, client *nex.Client, callID uint32, pidList []uint32)
+	server                                 *nex.Server
+	UpdateProfileHandler                   func(err error, client *nex.Client, callID uint32, profileData *MyProfile)
+	UpdateMiiHandler                       func(err error, client *nex.Client, callID uint32, mii *Mii)
+	UpdatePreferenceHandler                func(err error, client *nex.Client, callID uint32, unknown1 bool, unknown2 bool, unknown3 bool)
```

Comment:
According to [libctru](https://github.com/devkitPro/libctru/blob/98324d412ff6e11ed9ed0ee57e760f24f7f47fa2/libctru/include/3ds/services/frd.h#L114), the preferences are `publicMode`, `showGame` and `showPlayedGame`, in that order